### PR TITLE
Chore: @typescript-eslint + type imports

### PR DESCRIPTION
**Pull request by katspaugh**
## What it solves
Adds the typescript eslint plugin and a rule for it to automatically import types with the `type` keyword.

See https://github.com/safe-global/safe-react-gateway-sdk/pull/91 for context.

### CODE REVIEW

File: src/components/address-book/ExportDialog/index.tsx [hunk 1]
```diff
@@ -4,7 +4,8 @@ import Button from '@mui/material/Button'
 import Link from '@mui/material/Link'
 import Typography from '@mui/material/Typography'
 import { useCSVDownloader } from 'react-papaparse'
-import { SyntheticEvent, useMemo, type ReactElement } from 'react'
+import type { SyntheticEvent } from 'react'
+import { useMemo, type ReactElement } from 'react'
```
Comment (iamacook) [inline src/components/address-book/ExportDialog/index.tsx:8]: Doesn't seem to have worked here.

File: src/components/common/CopyButton/index.tsx [hunk 2]
```diff
@@ -1,4 +1,5 @@
-import React, { type ReactElement, ReactNode, type SyntheticEvent, useCallback, useState } from 'react'
+import type { ReactNode } from 'react'
+import React, { type ReactElement, type SyntheticEvent, useCallback, useState } from 'react'
```
Comment (iamacook) [inline src/components/common/CopyButton/index.tsx:2]: Here either.

**Review by iamacook — APPROVED** [review]
I'll approve it as it was fixed by the linter but there are multiple places where it didn't find "nested" imported types for some reason.

File: src/components/common/CopyButton/index.tsx (hunk 2 above)
Comment (iamacook) [reply src/components/common/CopyButton/index.tsx:2]: I won't mark the others.

Comment (katspaugh) [reply src/components/common/CopyButton/index.tsx:2]: Looks like it extracts them into their own import. Can't say I like it.

**Pull request merged**